### ISSUE TITLE
UrlHelper gets confused when multiple actions have the same name

### ISSUE THREAD

**Issue by blemasle** [aspnet/Mvc#3077]
UrlHelper seems to choose the first matched route whatever are the provided arguments when multiple routes have the same name.

````csharp
[Route("api/[controller]")]
public class PersonsController : Controller
{
	[HttpGet("{id}")]
	public Person Get(int id)
	{
		...
	}

	[HttpGet]
	public IEnumerable<Person> Get()
	{
		...
	}        
}
````

Now if you call `urlHelper.Action("Get", "Persons", new RouteValueDictionary(new { id = 19 }));` you'll get `/api/Persons?id=19` as a result. The only solution is to name the second route `GetAll` for instance. I do not recall this being the case on WebAPI 2.

Using Mvc 6.0.0-beta6 under Windows 7, IIS Express

**Comment by rynowak** [comment]
WebAPI didn't have link generation by action name https://msdn.microsoft.com/en-us/library/system.web.http.routing.urlhelper(v=vs.118).aspx

I agree with you though, I actually expected this to work the way you'd described and try the longer template `api/Persons/{id}` before the shorter one.

I think we got the sort wrong https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/Routing/InnerAttributeRoute.cs#L56  This might have better usability as an `OrderByDescending`

**Comment by blemasle** [comment]
Indeed, I mixed things up with named routes.

**Comment by ryanbrandenburg** [comment]
Fixed in #3404
